src/back/glsl/mod.rs

```diff
@@ -2941,33 +2932,55 @@ impl<'a, W: Write> Writer<'a, W> {
                     Mf::Determinant => "determinant",
                     // bits
                     Mf::CountLeadingZeros => {
+                        use std::f32::consts::LOG2_E;
+
                         match *ctx.info[arg].ty.inner_with(&self.module.types) {
                             crate::TypeInner::Vector { size, kind, .. } => {
                                 let s = back::vector_size_str(size);
 
+                                write!(self.out, "mix(")?;
+
                                 if let crate::ScalarKind::Uint = kind {
-                                    write!(self.out, "uvec{s}(ivec{s}(31) - findMSB(")?;
+                                    write!(self.out, "uvec{s}(")?;
+                                } else {
+                                    write!(self.out, "ivec{s}(")?;
+                                }
+
+                                write!(self.out, "vec{s}(31.0) - floor(log(vec{s}(")?;
+                                self.write_expr(arg, ctx)?;
+                                write!(self.out, ") + 0.5) * {LOG2_E})), ")?;
```

Comment:
Can we use `log2(x)` instead of `log(x) * LOG2E`?

Reply:
Yes, feel free to make whatever changes. I just transcribed it verbatim at the time